catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/InputTransformerIds.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.commons.io.FilenameUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class InputTransformerIds {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(InputTransformerIds.class);
+
+  private Path transformerFolder;
+
+  public InputTransformerIds() {
+    this(Paths.get("etc", "transformers"));
+  }
+
+  /** @param path path of directory containing {@link InputTransformerIds} json files */
+  public InputTransformerIds(Path path) {
+    transformerFolder = path;
+  }
+
+  public Set<String> getIds() {
+    File[] transformerFiles = transformerFolder.toFile().listFiles();
+    if (transformerFiles == null || transformerFiles.length == 0) {
+      LOGGER.debug("No Input Transformer json files found in {}", transformerFolder);
+      return Collections.emptySet();
+    }
+
+    Set<String> allTransformerIds = new HashSet<>();
+    Gson gson = new Gson();
+    for (File transformerFile : transformerFiles) {
+      if (!FilenameUtils.getExtension(transformerFile.toString()).equals("json")) {
+        LOGGER.debug("Skipping non-json file {}", transformerFile);
+        continue;
+      }
+
+      try (FileReader reader = new FileReader(transformerFile)) {
+
+        List<String> filesTransformerIds =
+            gson.fromJson(reader, new TypeToken<List<String>>() {}.getType());
+
+        allTransformerIds.addAll(filesTransformerIds);
+      } catch (IOException e) {
+        LOGGER.debug(
```

Comment:
If there is an error reading the file outside malformed content, then the route will start up normally.

Reply:
I'm starting to become concerned this could lead to unexpected behavior and may be difficult for someone to track down/fix after startup. I'm wondering if it would be better to continually fail rather than run in a partially broken state.

Reply:
I agree with passing constants with blueprint but I am not a fan of setters either. My preference would be another constructor that calls this one with hardcoded constants to be used by blueprint. The current constructor can then be annotated with `@VisibleForTesting`. 

I am a proponent of marking things final if they are not supposed to be modified during the life of an object. It simplifies review by making it clear it won't be modified, it simplifies concurrency since it is set at construction time, and more.

Reply:
If we block in initialization there will be a few oddities. They all seem like edge cases and not severe issues and still okay to continue forward.

1. MSF instantiates configurations serially. If one expected InputTransformer is not available, every CDM configuration will wait the maximum timeout.
2. If MSF is initializing a CDM configuration and expected InputTransformers are not available, shutdown will be delayed at most up to the InputTransformer wait timeout period. If there are multiple configurations, the first one initializing will timeout, and all remaining configurations will have destroy called on them.
3. If a CDM configuration is deleted while waiting, the config file will be deleted, but the initialization process will still continue till timeout and destroy will not be called. If the missing InputTransformer becomes available during this waiting period, but after the config file deletion, then the route will still be created, but immediately destroyed.